src/main/java/com/acrolinx/sidebar/utils/NamespaceResolver.java

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2023-present Acrolinx GmbH */
+package com.acrolinx.sidebar.utils;
+
+import java.util.Iterator;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+import org.w3c.dom.Document;
+
+public class NamespaceResolver implements NamespaceContext {
+  private final Document sourceDocument;
+
+  public NamespaceResolver(Document document) {
+    sourceDocument = document;
+  }
+
+  @Override
+  public String getNamespaceURI(String prefix) {
+    if (prefix.equals(XMLConstants.DEFAULT_NS_PREFIX)) {
+      return sourceDocument.lookupNamespaceURI(null);
+    } else {
+      return sourceDocument.lookupNamespaceURI(prefix);
+    }
+  }
+
+  @Override
+  public String getPrefix(String namespaceURI) {
+    return sourceDocument.lookupPrefix(namespaceURI);
+  }
+
+  @Override
+  public Iterator getPrefixes(String namespaceURI) {
+    return null;
```

Comment:
returning null is ugly, can we either return an empty Iterator or throw a `UnsupportedOperationException` or so?